scrapy/pipelines/files.py

```diff
@@ -241,6 +241,38 @@ def media_downloaded(self, response, request, info):
         referer = referer_str(request)
 
         if response.status != 200:
+
+            if response.status == 201:
+
+                try:
+                    loc = [response.headers['location']]
```

Comment:
`response.headers['location']` returns a single value. Do you need a list here?

If you do need to process a list, scrapy response [headers also have a `getlist(key)` method](https://github.com/scrapy/scrapy/blob/75cd056223a5a8da87a361aee42a541afcf27553/scrapy/http/headers.py#L50) (used for Cookies for example)

You can look at how `RedirectMiddleware` checks "Location" headers: https://github.com/scrapy/scrapy/blob/75cd056223a5a8da87a361aee42a541afcf27553/scrapy/downloadermiddlewares/redirect.py#L65